syftlib/src/main/java/org/openmined/syft/execution/Plan.kt

```diff
@@ -96,10 +95,14 @@ class Plan(val job: SyftJob, val planId: String) {
      * @return the absolute path of the file containing the TorchScript model.
      */
     private fun saveScript(filesDir: String, obj: com.google.protobuf.ByteString): String {
-        val fileWriter = FileWriter(filesDir, "torchscript_${planId}.pt")
-        fileWriter.outputStream().use {
+        val file = File( "$filesDir/torchscript_${planId}.pt")
+        file.parentFile?.let {
+            if (!it.exists()) it.mkdirs()
```

Comment:
This will fail when parent file is null so may be set it with elvis?

Reply:
It can't be null (`filesDir` is never null). The `?.let` is there because of `File` API.

Reply:
Alternatively,
```
        val parent = File(filesDir)
        if (parent.exists().not()) {
            parent.mkdirs()
        }
```

Reply:
No I meant parentFile might not be initialised. I agree filesDir is not null but it might not be set as parent. Last time I checked Files API initialising it without parent File explicitly causes this field to be null

Reply:
Done